source/crud/tests/server_2_6/write/findOneAndReplace.yml

```diff
@@ -0,0 +1,46 @@
+data:
+    - {_id: 1, x: 11}
+    - {_id: 2, x: 22}
+    - {_id: 3, x: 33}
+serverVersion: '2.6'
+
+tests:
+    -
+        description: "FindOneAndReplace when no documents match with upsert returning the document before modification"
+        operation:
+            name: findOneAndReplace
+            arguments:
+                filter: {_id: 4}
+                replacement: {x: 44}
+                projection: {x: 1, _id: 0}
+                sort: {x: 1}
+                upsert: true
+
+        outcome:
+            result: null
+            collection:
+                data:
+                    - {_id: 1, x: 11}
+                    - {_id: 2, x: 22}
+                    - {_id: 3, x: 33}
+                    - {_id: 4, x: 44}
```

Comment:
I can confirm that this needs to be a 2.6+ test due to [SERVER-5289](https://jira.mongodb.org/browse/SERVER-5289). Alternatively, you can add `{_id: 4}` to the replacement document to make it portable for 2.4.